include/utils/filo.h

```diff
@@ -0,0 +1,177 @@
+#ifndef _UTILS_FILO_H_
+#define _UTILS_FILO_H_
+
+/*
+ * Copyright (c) 2020 Abhishek Arora <abhiarora@gotomain.io>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdint.h>
+#include <stdbool.h>
+
+typedef struct {
+    void *buffer;
+    size_t top;
+    size_t elem_size;
+    size_t max_size;
+} filo_t;
+
+#define __FILO_DEF(name, type, sz) \
+    type name ## _buffer[sz]; \
+    filo_t name = { .buffer = name ## _buffer, \
+                     .top = 0, \
+                     .elem_size = sizeof(type),\
+                     .max_size = sz \
+                    }; \
+
+#define FILO_DEF(name, type, sz) \
+    __FILO_DEF(name, type, sz) \
+    int __## name ## _push(const type *elem) \
+    { \
+        return filo_push(&name, (void *)elem); \
+    } \
+    int __## name ## _pop(const type *elem) \
+    { \
+        return filo_pop(&name, (void *)elem, true); \
+    } \
+    int __## name ## _peek(const type *elem) \
+    { \
+        return filo_pop(&name, (void *)elem, false); \
+    } \
+    unsigned __## name ## _get_count(const type *elem) \
+    { \
+        return filo_get_count(&name); \
+    } \
+    void __## name ## _reset(const type *elem) \
+    { \
+        filo_reset(&name); \
+    } \
+
+#define FILO_EXTERN(name) \
+    extern filo_t name;
```

Comment:
This should also list the methods you newly created when calling filo_def here.(as you would in a header file). Otherwise calling them would cause warnings.